sklearn/manifold/t_sne.py

```diff
@@ -48,15 +52,55 @@ def _joint_probabilities(distances, desired_perplexity, verbose):
     # Compute conditional probabilities such that they approximately match
     # the desired perplexity
     conditional_P = _utils._binary_search_perplexity(
-        distances, desired_perplexity, verbose)
+        distances.astype(np.float), desired_perplexity, verbose)
```

Comment:
Maybe `sklearn.utils.validation.as_float_array` might be what you want here actually.